### PR TITLE
Allow to change internetworking model

### PR DESCRIPTION
**Pull request by jcarlosv**
This PR add configuation option to allow change how VM will be connected to container network. 

Allowed options are : `bridged` and `macvtap`
 
Depends on https://github.com/containers/virtcontainers/pull/528

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -219,6 +220,26 @@ func (h hypervisor) defaultBridges() uint32 {
...
+func (h hypervisor) internetworkingModel() (vc.NetInterworkingModel, error) {
+	var model vc.NetInterworkingModel
+	modelName := h.InterNetworkModel
+
+	if modelName == "" {
+		modelName = defaultInterNetworkingModel
+	}
+
+	ccLog.WithFields(
+		logrus.Fields{
```
Comment (jodh-intel) [inline config.go]: Nit: as you are only logging a single field, you could make this:

```
ccLog.WithField("model", modelName).Info("Setting internetworking model")
```

File: config.go [hunk 2]
```diff
@@ -219,6 +220,26 @@ func (h hypervisor) defaultBridges() uint32 {
...
+			"model": modelName,
+		}).Infof("Setting internetworking model")
```
Comment (jodh-intel) [inline config.go]: This can be `Info()` rather than `Infof()` here.

File: config.go [hunk 3]
```diff
@@ -219,6 +220,26 @@ func (h hypervisor) defaultBridges() uint32 {
...
+
+	err := model.SetModel(modelName)
+	if err != nil {
+		return vc.ModelDefault, err
+	}
+	return model, nil
```
Comment (jodh-intel) [inline config.go]: Missing blank line between closing brace and `return`.

File: config.go [hunk 4]
```diff
@@ -265,6 +286,10 @@ func newQemuHypervisorConfig(h hypervisor) (vc.HypervisorConfig, error) {
 	if err != nil {
 		return vc.HypervisorConfig{}, err
 	}
+	internetworkingModel, err := h.internetworkingModel()
```
Comment (jodh-intel) [inline config.go]: Missing blank line before this one.

File: config/configuration.toml.in [hunk 5]
```diff
@@ -81,6 +81,22 @@ disable_block_device_use = @DEFDISABLEBLOCK@
 # behaving as it would when running on bare metal.
 # 
 #disable_nesting_checks = true
+#
+# Internetworking model
+# Determines how the VM should be connected to the
+# the container network interface
+# Options:
+#
+# - bridged
```
Comment (jodh-intel) [inline config/configuration.toml.in]: I'd be tempted to indent these two options (and indent their descriptions further) for clarity:

```
# Options:
#
#   - bridged
#
#     Uses a linux bridge to interconnect
#     the container interface to the VM. ...
#          :
#
#   - macvtap
#
#     Used when the Container network
#          :
#

```

**Comment by jcvenegas** [comment]
Hi @jodh-intel  thanks for the review, changes applied.

**Comment by jodh-intel** [comment]
I noticed that the PR was raised using an alternative github id, but the commit authorship and sign-offs are as expected, so

lgtm.

Now, we just need to wait for https://github.com/containers/virtcontainers/pull/528/ to land and be re-vendored into the runtime.

File: config.go [hunk 6]
```diff
@@ -219,6 +220,24 @@ func (h hypervisor) defaultBridges() uint32 {
 	return h.DefaultBridges
 }
 
+func (h hypervisor) internetworkingModel() (vc.NetInterworkingModel, error) {
```
Comment (sboeuf) [inline config.go]: Could you please rename every `internetworkingModel` occurrences into `interNetworkingModel`.

**Comment by sboeuf** [comment]
@jcvenegas https://github.com/containers/virtcontainers/pull/528 has been merged. Could you please rebase this PR on latest origin/master and update the vendoring too ?

**Pull request closed without merging**